Common/Securities/Cash.cs

```diff
@@ -29,11 +29,18 @@ namespace QuantConnect.Securities
     /// </summary>
     public class Cash
     {
+        private decimal _conversionRate;
         private bool _isBaseCurrency;
         private bool _invertRealTimePrice;
 
         private readonly object _locker = new object();
 
+        /// <summary>
+        /// Event fired when this instance is updated
+        /// <see cref="AddAmount"/>, <see cref="SetAmount"/>, <see cref="Update"/>
+        /// </summary>
+        public event EventHandler Updated;
```

Comment:
no need to change :P -- I was originally thinking this would have some information in the event args, such as old value/new value - this would enable future code to be written incrementally and always ensure we're not re-computing things. For example, in this case consumers are likely to use the `ValueInAccountCurrency`, so multiple consumers would yield extra evaluations of the property.